### PR TITLE
[terra-table] Added setTimeout to resizeObserver

### PR DESCRIPTION
**Pull request by saket2403**
### Summary

**What was changed:**

Added setTimeout to resizeObserver

**Why it was changed:**
To fix **ResizeObserver loop completed with undelivered notifications** error

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10206 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-table/src/Table.jsx [hunk 1]
```diff
@@ -398,22 +400,26 @@ function Table(props) {
...
 
+  const handleResize = () => {
+    setTableHeight(tableRef.current.offsetHeight - 1);
+
+    const tableContainer = tableContainerRef.current;
+    setTableScrollable(tableContainer.scrollWidth > tableContainer.clientWidth
+                      || tableContainer.scrollHeight > tableContainer.clientHeight);
+  };
+
+  const debouncedHandleResize = LodashDebounce(handleResize, 300);
```
Comment (ShettyAkarsh) [inline packages/terra-table/src/Table.jsx:411]: what is this 300 hardcoded value ?

Comment (saket2403) [reply packages/terra-table/src/Table.jsx:411]: It is a 300ms delay before invoking the function.

File: packages/terra-table/src/Table.jsx [hunk 2]
```diff
@@ -35,6 +37,8 @@ const TableConstants = {
 
 const ROW_SELECTION_COLUMN_ID = 'table-rowSelectionColumn';
 
+const DEBOUNCE_TIMER = 300;
```
Comment (cm9361) [inline packages/terra-table/src/Table.jsx:40]: Do we want to debounce or throttle the resize logic?  With a debounce, you won't see any updates until you finish resizing.  Also, 300ms seems pretty high for a delay after the resize is completed.

File: packages/terra-table/src/Table.jsx [hunk 3]
```diff
@@ -398,22 +402,26 @@ function Table(props) {
...
-    });
+    const resizeObserver = new ResizeObserver(debouncedHandleResize);
 
     resizeObserver.observe(tableRef.current);
 
     return () => {
       resizeObserver.disconnect();
     };
-  }, [tableRef]);
+  }, [debouncedHandleResize]);
```
Comment (cm9361) [inline packages/terra-table/src/Table.jsx:424]: This logic is incorrect.  You are creating a new instance of debouncedHandleResize on every render.  So your useEffect will not trigger on every render and nullify the memoization.

**Review by sugan2416 — APPROVED** [review]

**Review by ShettyAkarsh — APPROVED** [review]

**Pull request merged**